composer/modules/server/services/ballerina-parser/src/main/java/org/ballerinalang/composer/service/ballerina/parser/service/BallerinaParserService.java

```diff
@@ -421,11 +428,14 @@ private JsonObject validateAndParse(BFile bFileRequest) throws InvocationTargetE
 
         BallerinaFile bFile;
         String programDir = "";
+        LSGlobalContext lsGlobalContext = new LSGlobalContext();
```

Comment:
Using a single global cache might cause issues when switching between multiple projects. Lets merge the PR and test it out, since current issue is blocker for composer and vscode.

Reply:
We'll go through a refactoring cycle over this API to generalize the completion context usage between composer as well as LS